demos/python_demos/audio_classification_demo/audio_classification_demo.py

```diff
@@ -0,0 +1,216 @@
+    args.add_argument("-d", "--device", type=str, default="CPU",
+                      help="Optional. Specify the target device to infer on; CPU, GPU, FPGA, HDDL or MYRIAD is"
+                           " acceptable. The sample will look for a suitable plugin for device specified. "
+                           "Default value is CPU")
+    args.add_argument('--labels', type=str, default=None,
+                      help="Optional. Labels mapping file")
+    args.add_argument('-sr', '--samplerate', type=int,
+                      help="Optional. Set samplerate for audio input")
+    args.add_argument('-ol', '--overlap', type=type_overlap, default=0,
+                      help='Optional. Set the overlapping between audio clip in samples or percent')
+
+    return parser.parse_args()
+
+
+class AudioSource:
+    def __init__(self, source, channels=2, samplerate=None):
+        self.source = source
+        self.samplerate = samplerate
+        self.channels = channels
+
+    def load(self):
+        samplerate, audio = read_wav(self.source, as_float=True)
+        audio = audio.T
+        if audio.shape[0] != self.channels:
+            raise RuntimeError("Audio has unsupported number of channels")
+        if self.samplerate:
+            if self.samplerate != samplerate:
+                audio = resample(audio, samplerate, self.samplerate)
+        else:
+            self.samplerate = samplerate
+
+        self.audio = audio
+
+    def chunks(self, size, hop=None, num_chunks=1):
+        if not hop:
+            hop = size
+        pos = 0
+
+        while pos + (num_chunks-1)*hop + size <= self.audio.shape[1]:
+            yield np.array([self.audio[:, pos+n*hop: pos+n*hop+size] for n in range(num_chunks)])
+            pos += hop*num_chunks
+
+
+def resample(audio, sample_rate, new_sample_rate):
+    duration = audio.shape[1] / float(sample_rate)
+    x_old = np.linspace(0, duration, audio.shape[1])
+    x_new = np.linspace(0, duration, int(duration*new_sample_rate))
+    data = np.array([np.interp(x_new, x_old, channel) for channel in audio])
+
+    return data
+
+
+def read_wav(file, as_float=False):
+    sampwidth_types = {
+        1: np.uint8,
+        2: np.int16,
+        3: np.int32,
+        4: np.int32
+    }
+    sampwidth_max = {
+        1: 255,
+        2: 2**15,
+        3: 2*23,
+        4: 2**31
+    }
+    with wave.open(file, "rb") as wav:
+        params = wav.getparams()
+        data = wav.readframes(params.nframes)
+        if sampwidth_types.get(params.sampwidth):
+            data = np.frombuffer(data, dtype=sampwidth_types[params.sampwidth])
+        else:
+            raise RuntimeError("Couldn't process file {}: unsupported sample width {}"
+                               .format(str(file), params.sampwidth))
+        data = np.reshape(data, (params.nframes, params.nchannels))
+        if as_float:
+            data = data / sampwidth_max[params.sampwidth]
```

Comment:
Would this work correctly for both 8-bit and 16-bit wave files? It seems that for 8-bit ones it would produce values in the [0;1] range, while for 16-bit files it would produce values in the [-1;1] range. But the network could only accept one of those.